[
    {
        "content": "<p><code>roc dev main.roc</code> (same as just <code>roc main.roc</code>) currently exits with code 0 if an expect failed <a href=\"https://github.com/roc-lang/roc/issues/7767\">#7767</a>. This is easy to fix but now the program exits as soon as one expect fails, with the old behavior we ran all expects and printed all failures. Is this early return fine or not?</p>",
        "id": 514876152,
        "sender_full_name": "Anton",
        "timestamp": 1745863950
    },
    {
        "content": "<p>Once the first expect fails, the program is in a broken state.</p>",
        "id": 514905182,
        "sender_full_name": "Brendan Hansknecht",
        "timestamp": 1745874906
    },
    {
        "content": "<p>As such, early failure makes sense</p>",
        "id": 514905197,
        "sender_full_name": "Brendan Hansknecht",
        "timestamp": 1745874912
    },
    {
        "content": "<p>Long term, this will be up to the platform</p>",
        "id": 514905215,
        "sender_full_name": "Brendan Hansknecht",
        "timestamp": 1745874920
    },
    {
        "content": "<p>Like basic webserver probably will log and fail just a single request in the long term</p>",
        "id": 514905264,
        "sender_full_name": "Brendan Hansknecht",
        "timestamp": 1745874949
    },
    {
        "content": "<p>Just linking to the PR for future searches: <a href=\"https://github.com/roc-lang/roc/pull/7768\">https://github.com/roc-lang/roc/pull/7768</a></p>",
        "id": 515079585,
        "sender_full_name": "Anton",
        "timestamp": 1745936920
    }
]